src/SonarQube.Client/Services/SonarQubeService2.cs

```diff
@@ -0,0 +1,205 @@
+        }
+
+        public async Task ConnectAsync(ConnectionInformation connection, CancellationToken token)
+        {
+            httpClient = new HttpClient(messageHandler)
+            {
+                BaseAddress = connection.ServerUri,
+                DefaultRequestHeaders =
+                {
+                    Authorization = AuthenticationHeaderFactory.Create(
+                        connection.UserName, connection.Password, connection.Authentication),
+                },
+            };
+
+            connected = true;
+
+            var versionResponse = await InvokeRequestAsync<IGetVersionRequest, string>(token);
+            sonarQubeVersion = Version.Parse(versionResponse);
+
+            var credentialResponse = await InvokeRequestAsync<IValidateCredentialsRequest, bool>(token);
+            if (!credentialResponse)
+            {
+                connected = false;
+                throw new InvalidOperationException("Invalid credentials");
+            }
+        }
+
+        public void Disconnect()
+        {
+            connected = false;
+            httpClient.Dispose();
+        }
+
+        public Uri GetProjectDashboardUrl(string projectKey)
+        {
+            EnsureIsConnected();
+
+            const string ProjectDashboardRelativeUrl = "dashboard/index/{0}";
+
+            return new Uri(httpClient.BaseAddress, string.Format(ProjectDashboardRelativeUrl, projectKey));
+        }
+
+        public async Task<IList<SonarQubeOrganization>> GetAllOrganizationsAsync(CancellationToken token) =>
+            await InvokeRequestAsync<IGetOrganizationsRequest, SonarQubeOrganization[]>(token);
+
+        public async Task<IList<SonarQubePlugin>> GetAllPluginsAsync(CancellationToken token) =>
+            await InvokeRequestAsync<IGetPluginsRequest, SonarQubePlugin[]>(token);
+
+        public async Task<IList<SonarQubeProject>> GetAllProjectsAsync(string organizationKey, CancellationToken token) =>
+            await InvokeRequestAsync<IGetProjectsRequest, SonarQubeProject[]>(
+                request =>
+                {
+                    request.OrganizationKey = organizationKey;
+                },
+                token);
+
+        public async Task<IList<SonarQubeProperty>> GetAllPropertiesAsync(CancellationToken token) =>
+            await InvokeRequestAsync<IGetPropertiesRequest, SonarQubeProperty[]>(token);
+
+        public async Task<SonarQubeQualityProfile> GetQualityProfileAsync(string projectKey, string organizationKey, SonarQubeLanguage language, CancellationToken token)
+        {
+            var qualityProfiles = await InvokeRequestAsync<IGetQualityProfilesRequest, SonarQubeQualityProfile[]>(
+                request =>
+                {
+                    request.ProjectKey = projectKey;
+                    request.OrganizationKey = organizationKey;
+                },
+                token);
+
+            // TODO: add language to the request configuration above
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this 'TODO' comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1135)